### PR TITLE
feature: add auth/dynamic get user position

### PR DESCRIPTION
**Pull request by nukkk97**
If we need email verification, we may need two more data for each user in database: isVerified and randomToken.
The time isn't that much so I'm considering if we should still implement it. For now, the basic auth without email verification is done.
Also, in "/restaurants", I change the original getPosition() api to watchPosition(), which allows us to dynamically get user's location. Now if you are walking, you should notice that the marker is also moving.

### CODE REVIEW

**Comment by yenting-biao** [comment]
The time isn't that much so I'm considering if we should still implement it. For now, the basic auth without email verification is done.
-> ok, we can do this in the future if we have time

File: public/map-and-location.png
Comment (yenting-biao) [inline public/map-and-location.png:1]: 這是什麼

**Review by yenting-biao — APPROVED** [review]

File: public/map-and-location.png
Comment (nukkk97) [reply public/map-and-location.png:1]: A great image.

**Comment by nukkk97** [comment]
Note that in Header.tsx,148: <AuthForm onCloseAuthModal={handleCloseAuthModal} />
Still need to be modified since I discard the modal props.

File: src/app/_components/AuthForm.tsx [hunk 1]
```diff
@@ -125,6 +108,7 @@ function AuthForm({ onCloseAuthModal }: AuthFormProps) {
                         登入
                     </button>
                 )}
+                 {!!props.error&&<p className="bg-red-100 text-red-600 text-center p-2">帳號或密碼錯誤，請重試！</p>}
```
Comment (togimoto) [inline src/app/_components/AuthForm.tsx:111]: 為什麼有兩個驚嘆號？

**Review by togimoto — CHANGES_REQUESTED** [review]

File: src/app/_components/AuthForm.tsx (hunk 1 above)
Comment (nukkk97) [reply src/app/_components/AuthForm.tsx:111]: It is just a safe way to convert to boolean

**Review by togimoto — APPROVED** [review]
Thanks!

**Pull request merged**